code/game/objects/machinery/floodlight.dm

```diff
@@ -358,27 +308,114 @@
 
-/obj/machinery/floodlight/landing/testroom
-	name = "Ambience Light"
-	desc = "A powerful light placed concealed on the base to provide better visibility."
-	density = 0
-	alpha = 0
-	resistance_flags = RESIST_ALL
+/obj/machinery/floodlight/colony/reset_light()
+	if(fswitch?.turned_on)
+		turn_light(null, TRUE)
+
+/obj/machinery/floodlight/colony/turn_light(mob/user, toggle_on, cooldown = 0, sparks)
+	if(!fswitch) //no master, should never happen
+		return
+	if(sparks)
+		var/datum/effect_system/spark_spread/spark_system = new
+		spark_system.set_up(5, 0, src)
+		spark_system.attach(src)
+		spark_system.start(src)
+	if(toggle_on && !on)
+		set_light(brightness_on)
+		fswitch.active_power_usage += FLOODLIGHT_TICK_CONSUMPTION
+		on = TRUE
+		icon_state = "floodon"
+		update_icon()
+	else if(!toggle_on && on)
+		set_light(0)
+		fswitch.active_power_usage -= FLOODLIGHT_TICK_CONSUMPTION
+		on = FALSE
+		icon_state = "floodoff"
+		update_icon()
+		if(cooldown > 0)
+			addtimer(CALLBACK(src, .proc/reset_light), cooldown)
+
+
+#undef FLOODLIGHT_TICK_CONSUMPTION
+
+/obj/machinery/colony_floodlight_switch
+	name = "Colony Floodlight Switch"
+	icon = 'icons/obj/machines/floodlight.dmi'
+	icon_state = "panelnopower"
+	desc = "This switch controls the floodlights surrounding the archaeology complex. It only functions when there is power."
+	density = FALSE
+	anchored = TRUE
+	var/turned_on = FALSE //has to be toggled in engineering
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 5
+	active_power_usage = 0
+	resistance_flags = UNACIDABLE|XENO_DAMAGEABLE
+	var/list/floodlist = list() // This will save our list of floodlights on the map
+
+/obj/machinery/colony_floodlight_switch/Initialize()
+	..()
+	. = INITIALIZE_HINT_LATELOAD
 
-/obj/machinery/floodlight/landing/testroom/Initialize(mapload, ...)
+/obj/machinery/colony_floodlight_switch/LateInitialize() //Populate our list of floodlights so we don't need to scan for them ever again
 	. = ..()
-	set_light(25)
+	for(var/obj/machinery/floodlight/colony/F in GLOB.machines)
+		floodlist += F
+		F.fswitch = src
+
+/obj/machinery/colony_floodlight_switch/update_icon()
+	if(machine_stat & NOPOWER)
+		icon_state = "panelnopower"
+	else if(turned_on)
+		icon_state = "panelon"
+	else
+		icon_state = "paneloff"
+
+/obj/machinery/colony_floodlight_switch/power_change()
+	..()
+	if(machine_stat & NOPOWER)
+		if(turned_on)
+			toggle_lights(FALSE)
+			turned_on = FALSE
+	update_icon()
+
+/obj/machinery/colony_floodlight_switch/proc/toggle_power()
+	turned_on = !turned_on
+
+
+/obj/machinery/colony_floodlight_switch/proc/toggle_lights(switch_on)
+	for(var/obj/machinery/floodlight/colony/F in floodlist)
+		F.turn_light(null, switch_on)
+
+/obj/machinery/colony_floodlight_switch/attack_paw(mob/living/carbon/monkey/user)
+	return src.attack_hand(user)
+
+/obj/machinery/colony_floodlight_switch/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+	if(!ishuman(user))
+		to_chat(user, "Nice try.")
+		return FALSE
```

Comment:
This should just be in attack_paw, no?

Reply:
No fucking clue what was the intent of the original coder, probably should remove